pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -212,6 +212,26 @@ func (r *hibernationReconciler) Reconcile(ctx context.Context, request reconcile
 		return reconcile.Result{}, r.updateClusterDeploymentStatus(cd, cdLog)
 	}
 
+	// Secret toggle allowing consumers to e.g. perform manual maintenance on machines without the
+	// controller attempting to reconcile desired CD.Spec.PowerState.
+	if paused, err := strconv.ParseBool(cd.Annotations[constants.PowerStatePauseAnnotation]); err == nil && paused {
+		cdLog.Info("skipping reconcile of PowerState as the powerstate-pause annotation is set")
+		changed := false
+		if cd.Status.PowerState != hivev1.ClusterPowerStateUnknown {
+			changed = true
+			cd.Status.PowerState = hivev1.ClusterPowerStateUnknown
+		}
+		msg := "the powerstate-pause annotation is set"
+		changed = r.setCDCondition(cd, hivev1.ClusterHibernatingCondition, hivev1.HibernatingReasonPowerStatePaused, msg, corev1.ConditionUnknown, cdLog) ||
+			changed
+		changed = r.setCDCondition(cd, hivev1.ClusterReadyCondition, hivev1.ReadyReasonPowerStatePaused, msg, corev1.ConditionUnknown, cdLog) ||
+			changed
```

Comment:
These conditions with PowerStatePaused is more informative than Unsupported (if someone erroneously puts the annotation true for Unsupported clusters - it is easy for them to be mistaken) - so I would prefer we move the annotation based decision making above the checking (and setting) of unsupported clusters

Reply:
I saw in the subsequent commit that you are aware this will happen - I'm curious - it didn't need any changes in the shouldStartMachines logic? Maybe a unit test to simulate the scenario would help?

Reply:
This is Unknown, not Unsupported. 

Yes, I needed to finagle the starting logic a bit. It still “worked” without it, but it was slow and the state transitions looked odd. 

Yes, I’ll see about adding a test.

Reply:
I meant the subsequent reconcile will set them to Unsupported [here](https://github.com/openshift/hive/blob/fa498060121355f35e620b09eb70e0ee4dc45a47/pkg/controller/hibernation/hibernation_controller.go#L201-L213)

Reply:
Ah, I get you now. I deliberately put the annotation check *after* the unsupported check because I want unsupported to take precedence. Does that make sense?

Reply:
Added a test to prove "unsupported" takes precedence over "paused".

Reply:
Oh. If I pause an unsupported cluster, the controller will thrash. That's not good.

Reply:
Fixed. Also reversed the precedence as discussed offline. And added UTs for unpausing: one to resume, one to hibernate.